# Autozygosity mapping using exome variant data and AutozygosityMapper

There are many ways by which a disease gene can be mapped in set of people who are affected by the same genetic condition. Of these autozygosity mapping is one of the simplest methods that requires a small number of patients; however, the parents of the patients must be consangieous. While microarray SNP genotype data is the gold standard for autozygosity mapping, many groups initially try to map the disease gene using the less reliable whole exome or whole genome variant data. This is becasue no matter how the patients are mapped, whole exome or whole genome variant data is need to detect the specicific deletereous variant and so by using the NGS variant data it map be possible to save money by not generating microarray SNP genotype data.

A number of command line, desktop or web applications have been developed for the detection of autozygous regions in inbred individuals. For this toutorial, we will use [___AutozygosityMapper___]([microarray SNP genotype data](https://teufelsberg.charite.de/AutozygosityMapper/human/index.html). The main reason this application was chosen is because it is a web site, consequently, you don't have to worry about security or compatibility issues when trying to run a program on your computer. 
